src/python/pants/backend/python/goals/run_python_source.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+
+from pants.backend.python.goals.run_helper import _create_python_source_run_request
+from pants.backend.python.target_types import PexEntryPointField, PythonSourceField
+from pants.backend.python.util_rules.pex_environment import PexEnvironment
+from pants.core.goals.run import RunFieldSet, RunRequest
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class PythonSourceFieldSet(RunFieldSet):
+    required_fields = (PythonSourceField,)
+
+    source: PythonSourceField
+
+
+@rule(level=LogLevel.DEBUG)
+async def create_python_source_run_request(
+    field_set: PythonSourceFieldSet, pex_env: PexEnvironment
+) -> RunRequest:
+    return await _create_python_source_run_request(
+        field_set.address,
+        entry_point_field=PexEntryPointField(field_set.source.value, field_set.address),
+        pex_env=pex_env,
+        # @TODO: How should we make this customizable?
```

Comment:
`pex_binary` had this as a field, which made sense (although doesn't lend itself to CLI-customizability very well).

A field is likely the only path forward, but the wording will need to be tweaked to make it make sense in the context of a `python_source`, `python_test`, or any other `PythonSourceField` target. The challenge is ensuring the wording doesn't pollute the user's understanding of the source file in unrelated environments (like when packaged or when tested)

Reply:
There's also the issue of the default. Ideally, I think it should be `False` to follow the user's element of least surprise. That _miiiiight_ break someone who was running the file as a `pex_binary`, which today runs in the sandbox.

We can possibly combine the field with a subsystem option which solves CLI-customizablity and we can use another mandatory-set so we can change the default.

Reply:
This is the only outstanding TODO

Reply:
I vote something like `run_goal_use_sandbox`, with a default to `False`. As discussed in Slack, our sandbox already was incomplete and we do not catch in-exhaustive dependencies lists. I think it's better to bias towards something that is more likely to Just Work how users expect.

Reply:
> I vote something like run_goal_use_sandbox

Specifically though, you're advocating for a field on the target, and not a `run` flag?